nextcloudappstore/core/api/v1/views.py

```diff
@@ -130,3 +133,30 @@ def get_object(self):
         release = get_object_or_404(release)
         self.check_object_permissions(self.request, release)
         return release
+
+
+class SessionObtainAuthToken(APIView):
+    """Modified version of rest_framework.authtoken.views.ObtainAuthToken.
+
+    Modified to return token based on SessionAuthentication, and not just data
+    sent for BasicAuthentication.
+    """
+
+    throttle_classes = (PostThrottle,)
+    permission_classes = (IsAuthenticated,)
```

Comment:
The old `post()` only looked at the BasicAuthentication data when determining what token should be sent.